src/utils/env/index.ts

```diff
@@ -102,15 +102,18 @@ export const getValueForContext = (
    */
   contextOrBranch: ContextOrBranch,
 ): EnvelopeEnvVarValue | undefined => {
-  const valueForContext = values.find((val) => {
-    const isSupportedContext = (SUPPORTED_CONTEXTS as readonly string[]).includes(contextOrBranch)
-    if (!isSupportedContext) {
-      // the "context" option passed in is actually the name of a branch
-      return val.context === 'all' || val.context_parameter === contextOrBranch
+  const isSupportedContext = (SUPPORTED_CONTEXTS as readonly string[]).includes(contextOrBranch)
+  if (!isSupportedContext) {
+    const valueMatchingAsBranch = values.find((val) => val.context_parameter === contextOrBranch)
```

Comment:
It didn't seem worth optimizing perf here since `values` will contain at most a few elements